lms/djangoapps/api_manager/users/tests.py

```diff
@@ -315,6 +315,141 @@ def test_user_list_get(self):
         if 'id' in response.data['results'][0]:
             self.fail("Dynamic field filtering error in UserSerializer")
 
+    def test_user_list_get_filters(self):
+        test_uri = self.users_base_uri
+
+        organizations = []
+        organizations.append(Organization.objects.create(display_name='ABC Organization'))
+        organizations.append(Organization.objects.create(display_name='XYZ Organization'))
+
+        course1 = CourseFactory.create(org='edX', number='CS101', run='2016_Q1')
+        course2 = CourseFactory.create(org='mit', number='CS101', run='2016_Q2')
+
+        users = []
+        users.append(UserFactory.create(first_name='John', last_name='Doe', email='john.doe@example.com'))
+        users.append(UserFactory.create(first_name='Micheal', last_name='Mcdonald', email='mic.mcdonald@example.com'))
+        users.append(UserFactory.create(first_name='Steve', last_name='Jobs', email='steve.jobs@edx.org'))
+
+        for user in users[:2]:
+            user.organizations.add(organizations[0])
+            CourseEnrollmentFactory.create(user=user, course_id=course1.id)
+
+        users[2].organizations.add(organizations[1])
+        CourseEnrollmentFactory.create(user=users[2], course_id=course2.id)
+
+        # fetch user data by exact name match
+        response = self.do_get('{}?name={}'.format(test_uri, 'John Doe'))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data['results']), 1)
+        self.assertEqual(response.data['results'][0]['full_name'], 'John Doe')
+
+        # fetch user data by partial name match
+        response = self.do_get('{}?name={}&match=partial'.format(test_uri, 'Jo'))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data['results']), 2)
+        self.assertEqual(response.data['results'][0]['full_name'], 'John Doe')
+        self.assertEqual(response.data['results'][1]['full_name'], 'Steve Jobs')
+
+        # fetch user data by exact email match
+        response = self.do_get('{}?email={}'.format(test_uri, 'steve.jobs@edx.org'))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data['results']), 1)
+        self.assertEqual(response.data['results'][0]['full_name'], 'Steve Jobs')
+        self.assertEqual(response.data['results'][0]['email'], 'steve.jobs@edx.org')
+
+        # fetch user data by partial email match
+        response = self.do_get('{}?email={}&match=partial'.format(test_uri, 'example.com'))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data['results']), 2)
+        self.assertEqual(response.data['results'][0]['full_name'], 'John Doe')
+        self.assertEqual(response.data['results'][1]['full_name'], 'Micheal Mcdonald')
+
+        # fetch user data by partial organization display_name match
+        response = self.do_get('{}?organization_display_name={}&match=partial'.format(test_uri, 'Organ'))
```

Comment:
I think we should test `organization_display_name` partial search here with `ABC` to make sure it returns 2 results instead of 3 which is maximum we can get.